File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/cpp/exercises/practice/kindergarten-garden/kindergarten_garden.cpp
#include "kindergarten_garden.h"
#include <stdexcept>

namespace kindergarten_garden {

const std::vector<std::string> Garden::children_ = {
    "Alice", "Bob", "Charlie", "David",
    "Eve", "Fred", "Ginny", "Harriet",
    "Ileana", "Joseph", "Kincaid", "Larry"
};

Garden::Garden(const std::string& diagram) {
    size_t newline_pos = diagram.find('\n');
    if (newline_pos == std::string::npos || diagram.find('\n', newline_pos + 1) != std::string::npos) {
        throw std::invalid_argument("Invalid diagram format");
    }
    
    std::string row1 = diagram.substr(0, newline_pos);
    std::string row2 = diagram.substr(newline_pos + 1);
    
    cups_.push_back({row1.begin(), row1.end()});
    cups_.push_back({row2.begin(), row2.end()});
}

std::vector<Plants> Garden::plants(const std::string& child) const {
    int child_index = -1;
    for (size_t i = 0; i < children_.size(); ++i) {
        if (children_[i] == child) {
            child_index = i;
            break;
        }
    }
    
    if (child_index == -1) {
        return {};
    }
    
    std::vector<std::string> result;
    for (const auto& row : cups_) {
        if (row.size() < static_cast<size_t>(2 * child_index + 1)) {
            throw std::out_of_range("Invalid child index");
        }
        result.push_back(convert(row[2 * child_index]));
        result.push_back(convert(row[2 * child_index + 1]));
    }
    
    return result;
}

Plants Garden::convert(char c) const {
    switch (c) {
        case 'G': return Plants::grass;
        case 'C': return Plants::clover;
        case 'R': return Plants::radishes;
        case 'V': return Plants::violets;
        default: throw std::invalid_argument("Invalid plant code");
    }
}

}  // namespace kindergarten_garden

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/java/exercises/practice/kindergarten-garden/src/main/java/KindergartenGarden.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class KindergartenGarden {
    private static final List<String> STUDENTS = Collections.unmodifiableList(
        Arrays.asList(
            "Alice", "Bob", "Charlie", "David",
            "Eve", "Fred", "Ginny", "Harriet",
            "Ileana", "Joseph", "Kincaid", "Larry"
        )
    );
    
    private final Map<String, List<Plant>> plantsMap = new HashMap<>();

    KindergartenGarden(String garden) {
        String[] rows = garden.split("\n");
        if (rows.length != 2) {
            throw new IllegalArgumentException("Garden must have exactly two rows");
        }
        
        List<String> row1Chunks = splitRow(rows[0]);
        List<String> row2Chunks = splitRow(rows[1]);
        
        for (int i = 0; i < STUDENTS.size(); i++) {
            String student = STUDENTS.get(i);
            String plantsStr = row1Chunks.get(i) + row2Chunks.get(i);
            List<Plant> plants = new ArrayList<>();
            for (char c : plantsStr.toCharArray()) {
                plants.add(Plant.fromCode(c));
            }
            plantsMap.put(student, plants);
        }
    }

    List<Plant> getPlantsOfStudent(String student) {
        return plantsMap.getOrDefault(student, Collections.emptyList());
    }

    private List<String> splitRow(String row) {
        List<String> chunks = new ArrayList<>();
        for (int i = 0; i < row.length(); i += 2) {
            int end = Math.min(i + 2, row.length());
            chunks.add(row.substring(i, end));
        }
        return chunks;
    }
}
